runtimes/kotlin-1.6/src/main/kotlin/io/openruntimes/kotlin/Server.kt

```diff
@@ -8,56 +8,173 @@ import java.io.StringWriter
 import java.io.PrintStream
 import java.io.PrintWriter
 import kotlinx.coroutines.runBlocking
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import java.io.UnsupportedEncodingException
+import com.google.gson.GsonBuilder
+import com.google.gson.Gson
+import java.lang.reflect.Method
+import java.util.HashMap
 
 suspend fun main() {
     Javalin
         .create()
         .start(3000)
-        .post("/") { runBlocking { execute(it) } }
+        .get("/*") { runBlocking { execute(it) } }
+        .post("/*") { runBlocking { execute(it) } }
+        .put("/*") { runBlocking { execute(it) } }
+        .delete("/*") { runBlocking { execute(it) } }
+        .patch("/*") { runBlocking { execute(it) } }
+        .options("/*") { runBlocking { execute(it) } }
+        .head("/*") { runBlocking { execute(it) } }
 }
 
 suspend fun execute(ctx: Context) {
-    if (ctx.header("x-internal-challenge").isNullOrBlank()) {
-        ctx.status(500).result("Unauthorized");
-        return;
+    var secret = ctx.header("x-open-runtimes-secret");
+    var serverSecret = System.getenv("OPEN_RUNTIMES_SECRET");
+
+    if(secret == null) {
+        secret = "";
+    }
+
+    if(serverSecret == null) {
+        serverSecret = "";
     }
-    if (ctx.header("x-internal-challenge") != System.getenv("INTERNAL_RUNTIME_KEY")) {
-        ctx.status(500).result("Unauthorized")
+
+    if (secret.equals("") || secret != serverSecret) {
```

Comment:
You can use `==` instead of `.equals` in Kotlin, applies to all usages in the PR